deeplake/core/storage/s3.py

```diff
@@ -153,12 +155,16 @@ def subdir(self, path: str, read_only: bool = False):
         return sd
 
     def _set(self, path, content):
-        self.client.put_object(
-            Bucket=self.bucket,
-            Body=content,
-            Key=path,
-            ContentType="application/octet-stream",  # signifies binary data
-        )
+        if len(content) >= 1 * GB:
+            stream = BytesIO(content)
+            self.client.upload_fileobj(stream, self.bucket, path)
+        else:
+            self.client.put_object(
```

Comment:
Is put_object enough faster than upload_fileobj that it's worth having the else clause still? Or can we always use upload_fileobj ?

Reply:
`put_object` is direct mapping to the PutObject api call and considered as a default way to upload object. I'm not sure about the performance differences, but I see `upload_fileobj` as an edge case scenario and not as a default upload method.